include/exiv2/slice.hpp

```diff
@@ -426,9 +372,9 @@ namespace Exiv2
         template <typename storage_type>
         struct PtrSliceStorage
         {
-            typedef typename remove_cv<typename remove_pointer<storage_type>::type>::type value_type;
-            typedef value_type* iterator;
-            typedef const value_type* const_iterator;
+            using value_type = typename std::remove_cv<typename std::remove_pointer<storage_type>::type>::type;
+            using iterator = value_type*;
+            using const_iterator = const iterator;
```

Comment:
This is the problem: `const_iterator` is not `const value_type*` as one would think (I certainly did), but it's a constant `iterator` i.e. a **constant** pointer and not a `const value_type*` which is a pointer to a constant.
```suggestion
            using const_iterator = const value_type*;
```